frontend/src/host_orchestrator/orchestrator/validation.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package orchestrator
+
+import (
+	"fmt"
+	"regexp"
+
+	apiv1 "github.com/google/android-cuttlefish/frontend/src/host_orchestrator/api/v1"
+	"github.com/google/android-cuttlefish/frontend/src/liboperator/operator"
+)
+
+type SnapshotIDValidator string
+
+var snapshotIDRegex = regexp.MustCompile(`^([a-z0-9\-_]+)$`)
+
+func (v SnapshotIDValidator) Validate() error {
+	id := string(v)
+	if !snapshotIDRegex.MatchString(id) {
+		return operator.NewBadRequestError(
+			"invalid snapshot id value", fmt.Errorf("%s does not match %s", id, snapshotIDRegex))
+	}
+	return nil
+}
+
+type CreateSnapshotRequestValidator struct {
+	*apiv1.CreateSnapshotRequest
+}
+
+func (v CreateSnapshotRequestValidator) Validate() error {
+	if v.SnapshotID != "" {
+		if err := SnapshotIDValidator(v.SnapshotID).Validate(); err != nil {
```

Comment:
this can also be a function: `ValidateSnapshotID(v.SnapshotID)`

Reply:
Done.